### PR TITLE
Friends list 

### PR DESCRIPTION
**Pull request by martinalong**
- Friends list component
- Refactored button "full" to be a prop rather than a separate component (left the option to use <ButtonPrimaryFull /> since some people may already be using it, but updated the documentation to reflect the other format instead so we can be consistent across our components)
- Correction to settings component to remove "auto approve deals" setting since it isn't a part of powergate
- Correction to documentation for storage deals component so it is compatible with async fetching
- Edited Powergate Host URL to retrieve current value from node_common/constants.js instead of hard coding it. This way we no longer need to change each example everytime the host URL changes

Up next for friends features:
- Seeing storage/retrieval deals from friends

image

### CODE REVIEW

File: components/system/components/fragments/TableComponents.js [hunk 1]
```diff
@@ -100,7 +100,6 @@ const STYLES_COLUMN = css`
   align-items: flex-start;
   justify-content: space-between;
   align-self: stretch;
-  min-width: 10%;
```
Comment (jimmylee) [inline components/system/components/fragments/TableComponents.js:103]: `min-width: 10%` is a style added to catch edge cases on safari and mobile safari, did you test in those browsers?

File: components/system/modules/FriendsList.js [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+  ${STYLES_BUTTON}
+  color: ${Constants.system.brand};
+`;
+
+const STYLES_REJECT_BUTTON = css`
+  ${STYLES_BUTTON}
+  color: ${Constants.system.black};
+`;
+
+const centerLeftStyle = {
```
Comment (jimmylee) [inline components/system/modules/FriendsList.js:38]: Don't break the naming trend for these. use `STYLES_INLINE_`

File: components/system/modules/FriendsList.js [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+                style: centerRightStyle,
+                contentStyle: { padding: "0px" },
+              },
+            ],
+            rows: this.props.data.requests.map((each) => {
+              return {
+                id: each.user,
+                user: (
+                  <a
+                    href={"/" + each.user}
```
Comment (jimmylee) [inline components/system/modules/FriendsList.js:101]: {\`/${each.user}\`}

File: components/system/modules/FriendsList.js [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+                      src={each.img || genericImg}
+                      alt=""
+                      style={{ width: "40px" }}
+                    />
+                  </div>
+                ),
+                children: (
+                  <div>
+                    <div
+                      style={{
```
Comment (jimmylee) [inline components/system/modules/FriendsList.js:200]: Why isn't this just a class?

File: pages/experiences/filecoin-settings.js [hunk 5]
```diff
@@ -52,13 +54,9 @@ const EXAMPLE_CODE = `import * as React from "react";
 import { FilecoinSettings } from "slate-react-system";
 import { createPow } from "@textile/powergate-client";
 
-const PowerGate = createPow({ host: "http://pow.slate.textile.io:6002" });
+const PowerGate = createPow({ host: "${POWERGATE_HOST}" });
```
Comment (jimmylee) [inline pages/experiences/filecoin-settings.js:57]: People don't have access to `POWERGATE_HOST` outside of Slate, so please change this back.

File: components/system/modules/FriendsList.js [hunk 6]
```diff
@@ -0,0 +1,245 @@
...
+
+  _handleClick = (e) => {
+    this.setState({ selectedRowId: e.target.value });
+  };
+
+  render() {
+    return (
+      <React.Fragment>
+        <div
+          style={{ fontSize: Constants.typescale.lvl2, marginBottom: "8px" }}
```
Comment (jimmylee) [inline components/system/modules/FriendsList.js:64]: Just make these classes? It will be faster than computing inline styles performance wise.

File: components/system/components/fragments/TableComponents.js (hunk 1 above)
Comment (martinalong) [reply components/system/components/fragments/TableComponents.js:103]: Ahh good to know. No worries, I can leave it in there!
